pkg/util/stop/stopper.go

```diff
@@ -240,6 +241,17 @@ func (s *Stopper) refuseRLocked() bool {
 	return s.mu.stopping || s.mu.quiescing
 }
 
+// OnQuiesce is like AddCloser, but invokes on quiesce.
```

Comment:
This needs a comment similar to `AddCloser` which says that f is called synchronously if stopper is already quiescing/stopping.